src/pages/SearchPage/SearchPage.tsx

```diff
@@ -0,0 +1,93 @@
+import { useState, useEffect } from 'react';
+import {
+  CombinedSearchResults,
+  PostSearchResult,
+  UserSearchResult
+} from './components';
+import { BUTTON_LABELS, BUTTON_NAMES } from './constants';
+import { SearchIcon } from '~/assets';
+import { Button, Input } from '~/components/common';
+import { Header } from '~/components/domain';
+import { useForm } from '~/hooks';
+import { useSearchAll } from '~/services/searchService';
+
+type SelectedQuery = 'all' | 'post' | 'user';
+
+const SearchPage = () => {
+  const [query, handleQuery] = useForm();
+
+  const [debouncedQuery, setDebouncedQuery] = useState(query);
+  const [selectedQuery, setSelectedQuery] = useState<SelectedQuery>('all');
+
+  const {
+    data: { parsedPostResults, userResults }
+  } = useSearchAll({ query: debouncedQuery });
+
+  const activeButtonStyle =
+    'after:absolute after:-bottom-1 after:left-0 after:h-[5px] after:w-full after:content-[""] after:bg-main-lighten after:rounded-md';
+
+  const components = {
+    all: CombinedSearchResults,
+    post: PostSearchResult,
+    user: UserSearchResult
+  };
+
+  const Component = components[selectedQuery];
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    const { name } = event.target as HTMLButtonElement;
```

Comment:
이렇게 하면 타입 단언을 안해도 괜찮을 것 같아요..!

```suggestion
    const { name } = event.currentTarget;
```